components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/signup/UserSelfRegistrationManager.java

```diff
@@ -794,6 +842,29 @@ private UserStoreManager getUserStoreManager(User user) throws IdentityRecoveryE
         }
     }
 
+    /**
+     * Validate context username and tenant with the stored user's username and tenant domain..
+     *
+     * @param user User
+     * @throws IdentityRecoveryException Invalid Username/Tenant.
+     */
+    private void validateUser(User user) throws IdentityRecoveryException {
+
+        String contextUsername = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
```

Comment:
IMO validating the tenant domain before validating the username would be better

Reply:
Addressed in https://github.com/wso2-extensions/identity-governance/pull/414/commits/b5f0c45e2ce7cb40a24f7a051acc52455755cce4